static/app/components/globalDrawer/index.tsx

```diff
@@ -0,0 +1,109 @@
+import {createContext, useCallback, useContext, useEffect, useRef, useState} from 'react';
+import {AnimatePresence} from 'framer-motion';
+
+import ErrorBoundary from 'sentry/components/errorBoundary';
+import {DrawerBody, DrawerPanel} from 'sentry/components/globalDrawer/components';
+import type {
+  DrawerConfig,
+  DrawerContext as TDrawerContext,
+} from 'sentry/components/globalDrawer/types';
+import {t} from 'sentry/locale';
+import {useHotkeys} from 'sentry/utils/useHotkeys';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOnClickOutside from 'sentry/utils/useOnClickOutside';
+
+const DrawerContext = createContext<TDrawerContext>({
+  openDrawer: () => {},
+  closeDrawer: () => {},
+});
+
+export function GlobalDrawer({children}) {
+  const location = useLocation();
+  const [drawerConfig, setDrawerConfig] = useState<DrawerConfig | undefined>();
+  // If no 'drawerConfig' is set, the global drawer is closed.
+  const isDrawerOpen = drawerConfig !== undefined;
+  const openDrawer = useCallback<TDrawerContext['openDrawer']>(
```

Comment:
I assume this means that if a drawer is already open, calling open with a different config will just override and render a different drawer? From a visual experience pov, do we want to allow that or should we require for the current drawer to be closed first?

Reply:
yeah this would just swap the content completely out.

If we want the effect of one drawer closing and another opening we would need to render elements based on some unique id so that both get rendered at once and one animates out while the other animates in